tools/jvm/src/main/scala/org/scalajs/core/tools/linker/backend/closure/ClosureLinkerBackend.scala

```diff
@@ -74,15 +74,14 @@ final class ClosureLinkerBackend(
     val builder = new ClosureAstBuilder(config.relativizeSourceMapBase)
 
     // Build Closure IR
-    logger.time("Emitter (create Closure trees)") {
-      emitter.emit(unit, builder, logger)
+    val coreJSLib = logger.time("Emitter (create Closure trees)") {
+      emitter.emitForClosure(unit, builder, logger)
```

Comment:
I have considered doing that many times over the years. It would be a pretty huge endeavor.  And it would really make the content of `scalajsenv` significantly harder to read and write. I am not sure it would be a win overall.

At least now we're honest with ourselves that the `ClosureLinkerBackend` is a "friend" of `Emitter`. Before it was already the case, but we tried to pretend it was fine with `emitAll`/`emitPrelude`/etc.

We can reconsider later. If at all, IMO it should be done after this PR. IMO our API is cleaner now than before already. I wouldn't call what we have now an "escape hatch" any more than what we had before. And at least now it's `private[backend]`. Before it was public.